custom_components/neviweb130/climate.py

```diff
@@ -1916,35 +1977,33 @@ def hvac_modes(self):
             return SUPPORTED_HVAC_MODES
 
     @property
-    def current_temperature(self) -> float:
+    def current_temperature(self) -> float | None:
         """Return the room current temperature."""
         return self._cur_temp
 
     @property
-    def target_temperature(self) -> float:
+    def target_temperature(self) -> float | None:
         """Return the temperature we try to reach less Eco Sinope dr_setpoint delta."""
-        if self._target_temp is not None:
-            temp = self._target_temp + self._drsetpoint_value
+        if self.hvac_mode == HVACMode.COOL:
+            return self.target_temperature_high
         else:
-            temp = 0
-        if temp < self._min_temp:
-            return self._min_temp
-        if temp > self._max_temp:
-            return self._max_temp
-        return temp
+            return self.target_temperature_low
+
+    @property
+    def target_temperature_low(self) -> float | None:
+        """Return the heating temperature we try to reach less Eco Sinope dr_setpoint delta."""
+        if self._target_temp is None:
+            return None
+
+        return self._target_temp + self._drsetpoint_value
 
     @property
-    def target_cool_temperature(self) -> float:
+    def target_temperature_high(self) -> float | None:
         """Return the cooling temperature we try to reach."""
-        if self._target_cool is not None:
-            temp = self._target_cool
-        else:
-            temp = 0
-        if temp < self._cool_min:
-            return self._cool_min
-        if temp > self._cool_max:
-            return self._cool_max
-        return temp
+        if self._target_cool is None:
+            return None
+
+        return self._target_cool + self._drsetpoint_value
```

Comment:
I am not sure if `self._drsetpoint_value` should affect both `self._target_cool` and `self._target_temp` or only `self._target_temp`, or if it should be a minus (`-`)